Firebase/Database/Login/FAuthTokenProvider.m

```diff
@@ -16,65 +16,50 @@
 
 #import "FAuthTokenProvider.h"
 #import "FUtilities.h"
+#import "FIRAppInternal.h"
 #import "FIRLogger.h"
 #import "FIRDatabaseQuery_Private.h"
 #import "FIRNoopAuthTokenProvider.h"
 
-static NSString *const FIRAuthStateDidChangeInternalNotification = @"FIRAuthStateDidChangeInternalNotification";
-static NSString *const FIRAuthStateDidChangeInternalNotificationTokenKey = @"FIRAuthStateDidChangeInternalNotificationTokenKey";
-
-
-/**
- * This is a hack that defines all the methods we need from FIRFirebaseApp. At runtime we use reflection to get an
- * actual instance of FIRFirebaseApp. Since protocols don't carry any runtime information and selectors are invoked
- * by name we can write code against this protocol as long as the method signatures of FIRFirebaseApp don't change.
- */
-@protocol FIRFirebaseAppLike <NSObject>
-
-- (void)getTokenForcingRefresh:(BOOL)forceRefresh withCallback:(void (^)(NSString *_Nullable token, NSError *_Nullable error))callback;
-
-@end
-
-
 /**
  * This is a hack that defines all the methods we need from FIRAuth.
  */
 @protocol FIRFirebaseAuthLike <NSObject>
```

Comment:
Erg...  I forgot about this.  It's another hack that will need to be copied anywhere else somebody tries to consume this notification.  Maybe we should add a FIRAuthStateDidChangeInternalNotificationAppKey that contains the FIRApp instance in the notification object?